### PR TITLE
Soumyo/ec2 transit gateway attachment

### PR DESCRIPTION
**Pull request by soumyo13**
### Description

`ec2 transit gateway attachment` InSpec audit resource to test properties of a transit gateway attachment

### Issues Resolved
This is a new resource. There are no issues to be fixed.

### Check List
Please fill box or appropriate ([x]) or mark N/A.
- [x] New functionality includes integration tests/controls
- [x] New Terraform resources
- [x] Documentation provided or updated for resources 
- [x] All Integration Tests pass
- [x] All Unit Tests pass
- [x] `rake lint` passes
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: libraries/aws_ec2_transit_gateway_attachments.rb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  end
+
+  def fetch_data
+    transit_gateway_attachment_rows = []
+    pagination_options = {}
+    loop do
+      catch_aws_errors do
+        @api_response = @aws.compute_client.describe_transit_gateway_attachments(pagination_options)
+      end
+      return [] if !@api_response || @api_response.empty?
```
Comment (sathish-progress) [inline libraries/aws_ec2_transit_gateway_attachments.rb:42]: this should return `transit_gateway_attachment_rows`

File: test/integration/verify/controls/aws_ec2_transit_gateway_attachment.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+aws_resource_owner_id = attribute(:aws_resource_owner_id, default: '', description: 'The ID of the AWS account that owns the resource.')
+aws_resource_type = attribute(:aws_resource_type, default: '', description: 'The resource type. Valid values are vpc | vpn | direct-connect-gateway | peering | connect.')
+aws_resource_id = attribute(:aws_resource_id, default: '', description: 'The ID of the resource.')
+aws_state = attribute(:aws_state, default: '', description: 'The state of the attachment. Valid values are available | deleted | deleting | failed | failing | initiatingRequest | modifying | pendingAcceptance | pending | rollingBack | rejected | rejecting.')
+aws_creation_time = attribute(:aws_creation_time, default: '', description: 'The creation time of the transit gateway.')
+aws_association_transit_gateway_route_table_id = attribute(:aws_association_, default: '', description: 'The ID of the route table for the transit gateway.')
+aws_association_state = attribute(:aws_association_, default: '', description: 'The state of the attachment. Valid values are | associating | associated | disassociating | disassociated |')
```
Comment (sathish-progress) [inline test/integration/verify/controls/aws_ec2_transit_gateway_attachment.rb:10]: they both use same key! also I don't see these attributes into outputs.tf nor as variables

File: test/integration/verify/controls/aws_ec2_transit_gateway_attachments.rb [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+aws_resource_owner_id = attribute(:aws_resource_owner_id, default: '', description: 'The ID of the AWS account that owns the resource.')
+aws_resource_type = attribute(:aws_resource_type, default: '', description: 'The resource type. Valid values are vpc | vpn | direct-connect-gateway | peering | connect.')
+aws_resource_id = attribute(:aws_resource_id, default: '', description: 'The ID of the resource.')
+aws_state = attribute(:aws_state, default: '', description: 'The state of the attachment. Valid values are available | deleted | deleting | failed | failing | initiatingRequest | modifying | pendingAcceptance | pending | rollingBack | rejected | rejecting.')
+aws_creation_time = attribute(:aws_creation_time, default: '', description: 'The creation time of the transit gateway.')
+aws_association_transit_gateway_route_table_id = attribute(:aws_association_, default: '', description: 'The ID of the route table for the transit gateway.')
+aws_association_state = attribute(:aws_association_, default: '', description: 'The state of the attachment. Valid values are | associating | associated | disassociating | disassociated |')
```
Comment (sathish-progress) [inline test/integration/verify/controls/aws_ec2_transit_gateway_attachments.rb:10]: same as above here

File: test/integration/verify/controls/aws_ec2_transit_gateway_attachments.rb [hunk 4]
```diff
@@ -0,0 +1,27 @@
+aws_transit_gateway_attachment_id = attribute(:aws_transit_gateway_attachment_id, default: '', description: 'The IDs of the attachments.')
+aws_transit_gateway_id = attribute(:aws_transit_gateway_id, default: '', description: 'The ID of the transit gateway.')
+aws_transit_gateway_owner_id = attribute(:aws_transit_gateway_owner_id, default: '', description: 'The ID of the AWS account that owns the transit gateway.')
+aws_resource_owner_id = attribute(:aws_resource_owner_id, default: '', description: 'The ID of the AWS account that owns the resource.')
+aws_resource_type = attribute(:aws_resource_type, default: '', description: 'The resource type. Valid values are vpc | vpn | direct-connect-gateway | peering | connect.')
```
Comment (sathish-progress) [inline test/integration/verify/controls/aws_ec2_transit_gateway_attachments.rb:5]: where are these attributes defined ?

**Comment by sathish-progress** [comment]
over all this PR is good. Just missing some unit test attrs

**Comment by sathish-progress** [comment]
also missing update to readme file here

File: test/integration/verify/controls/aws_ec2_transit_gateway_attachments.rb (hunk 4 above)
Comment (soumyo13) [reply test/integration/verify/controls/aws_ec2_transit_gateway_attachments.rb:5]: done

**Comment by soumyo13** [comment]
> also missing update to readme file here

done

File: libraries/aws_ec2_transit_gateway_attachments.rb (hunk 1 above)
Comment (soumyo13) [reply libraries/aws_ec2_transit_gateway_attachments.rb:42]: done

**Review by ppradhan9 — COMMENTED** [review]
Please resolve conflicts

**Review by Nirbhay1997 — APPROVED** [review]
LGTM

**Pull request merged**